### ISSUE TITLE
Adapter Phantom: signMessage doesn't work as expected

### ISSUE THREAD

**Issue by macols77** [Web3Auth/Web3Auth#156]
I've been implementing Web3Auth in my app. All worked smoothly but while I was trying to use Phantom as an external wallet it wasn't working. I've been debugging and I'll show you why and how I manage to 'hack fix' the bug.

When I call signMessage from my UI I'm calling your code inside the solana provider package.

This function from solana-provider/src/solanaWallet:
![image](https://user-images.githubusercontent.com/804054/166160672-63e4439a-b837-4557-9a8e-d9ff9b41e376.png)

The problem was that when I was calling that method with my Phantom wallet connected I was receiving this output:
![image](https://user-images.githubusercontent.com/804054/166160847-aa6d2973-fd76-4e1e-bfcb-3e55b9fd7aa0.png)

I wasn't receiving a Uint8Array as expected. So I started to check the code behind window.solana.signMessage which is the code used by the solana-adapter-* sdk. I found this:
![image](https://user-images.githubusercontent.com/804054/166160998-a097caad-ebba-45da-96d6-e2dab518ea51.png)

That code basically first request the user to sign the message and after receiving the signature, it's decoded from base58 to Uint8Array.

So in my code to be able to handle the Phantom response using Web3Auth I've done the following:
![image](https://user-images.githubusercontent.com/804054/166161203-98d5fbff-b957-4c3e-9714-f1f09b2441cd.png)

It seems to be an incompatibility in the interface between phantom and web3auth's wallet.

**Comment by chaitanyapotti** [comment]
This is fixed in 1.0.0-alpha.0

**Comment by macols77** [comment]
I've updated my web3auth dependencies to 1.0.0-alpha.0 and It doesn't work. However, the error is different. Now I receive the following info from the signMessage method:

![image](https://user-images.githubusercontent.com/804054/168469548-ef615ffc-9581-40a2-bf88-abfb2e04c9fa.png)

Data is the signature as a array of numbers.

So I had to update the code as following:

![image](https://user-images.githubusercontent.com/804054/168469569-2e965ccc-a49d-4d52-9283-3f27427d9b26.png)
